tools/image_processing/yolo/preprocessing.py

```diff
@@ -0,0 +1,73 @@
+import argparse
+import json
+import os
+import shutil
+
+from sklearn.model_selection import train_test_split
+
+
+def get_basename(f):
+    return os.path.splitext(os.path.basename(f))[0]
+
+
+def pair_files(images_dir, labels_dir):
+
+    img_files = [f for f in os.listdir(images_dir)]
+    lbl_files = [f for f in os.listdir(labels_dir)]
+
+    image_dict = {get_basename(f): f for f in img_files}
+    label_dict = {get_basename(f): f for f in lbl_files}
+
+    keys = sorted(set(image_dict) & set(label_dict))
+
+    return [(image_dict[k], label_dict[k]) for k in keys]
+
+
+def copy_pairs(pairs, image_src, label_src, image_dst, label_dst):
+    os.makedirs(image_dst, exist_ok=True)
+    os.makedirs(label_dst, exist_ok=True)
+    for img, lbl in pairs:
+        shutil.copy(os.path.join(image_src, img), os.path.join(image_dst, img))
+        shutil.copy(os.path.join(label_src, lbl), os.path.join(label_dst, lbl))
+
+
+def write_yolo_yaml(output_dir, meta_json):
+    with open(meta_json, 'r') as f:
+        meta = json.load(f)
+
+    yolo_yaml_path = os.path.join(output_dir, "yolo.yml")
+    with open(yolo_yaml_path, 'w') as f:
+        f.write(f"path: {output_dir}\n")
+        f.write("train: train\n")
+        f.write("val: valid\n")
+        f.write("test: test\n")
+        f.write("\n")
+        f.write(f"nc: {meta['training_params']['num_class']}\n")
+        f.write("names: ['dataset']\n")
```

Comment:
the `names` should always contain `dataset` keyword?

Reply:
here https://github.com/bgruening/galaxytools/pull/1634#discussion_r2133878510